Add customization to create WebLogic domain home scripts

doxiao opened the pull request:
1) Moved out the scripts that are previously inside the create domain configmap into separate files; those files can be replaced by customers' scripts.
2) Added inputs properties to customize the behavior
3) Renamed podDomainRootDir to domainPVMountPath
4) Renamed weblogicImage to image, and weblogicImagePullSecretName to imagePullSecretName
5) Use env variables instead of string substitution on in the build-in create domain scripts.

(note #4 and #5 addressed review comments)

@rjeberhard @moreaut @tbarnes-us

moreaut commented on kubernetes/samples/scripts/create-weblogic-domain/domain-home-on-pv/create-weblogic-sample-domain-job-template.yaml:
```diff
@@ -1,264 +1,5 @@
 # Copyright 2017, 2018, Oracle Corporation and/or its affiliates.  All rights reserved.
```
Don't include 2017 in any of your copyrights?

doxiao replied:
Sure

moreaut commented on kubernetes/samples/scripts/create-weblogic-domain/domain-home-on-pv/create-weblogic-sample-domain.sh:
```diff
@@ -348,16 +365,66 @@ function createYamlFiles {
+  do
+    local outputFile=$fname
+    sed -i -e "s:%NAMESPACE%:$namespace:g" ${outputFile}
+    sed -i -e "s:%WEBLOGIC_CREDENTIALS_SECRET_NAME%:${weblogicCredentialsSecretName}:g" ${outputFile}
```
Should you match the operator helm chart?  It uses 'image', 'imagePullPolicy' and 'imagePullSecrets' (where the secret is the actual list of image pull secrets, instead of just a secret name, which would mean changing the same to parse yaml, instead of simple string values).

doxiao replied:
I guess we should if we want to have a consistent/better way of doing the same thing. I'll take a look at the operator helm chart.

doxiao replied:
I am changing the property names to be consistent with what we have in the operator helm charts:  weblogicImage -> image, and weblogicImagePullSecretName -> imagePullSecretName. Having discussed with Tom offine, we decided to leave the sample to only support one image pull secret.

moreaut commented on kubernetes/samples/scripts/create-weblogic-domain/domain-home-on-pv/create-weblogic-sample-domain.sh:
```diff
@@ -348,16 +365,66 @@ function createYamlFiles {
+    sed -i -e "s:%DOMAIN_PVC_NAME%:${persistentVolumeClaimName}:g" ${outputFile}
+    sed -i -e "s:%DOMAIN_ROOT_DIR%:${domainPVMountPath}:g" ${outputFile}
+    sed -i -e "s:%DOMAIN_LOGS_DIR%:${domainPVMountPath}/logs:g" ${outputFile}
+    sed -i -e "s:%CREATE_DOMAIN_SCRIPT_DIR%:${createDomainScriptsMountPath}:g" ${outputFile}
```
Do you really want different vars for the create domain script directory and name, instead of just a single pathname?

doxiao replied:
There can be multiple shell scripts involved in creating the domain home. We need to know the script that the job needs to run.

moreaut replied:
ok

doxiao commented on kubernetes/samples/scripts/create-weblogic-domain/domain-home-on-pv/create-weblogic-sample-domain.sh:
```diff
@@ -348,16 +365,66 @@ function createYamlFiles {
+
+  # create the configmap and label it properly
+  kubectl create configmap ${domainUID}-create-weblogic-sample-domain-job-cm -n $namespace --from-file $externalFilesTmpDir
+  kubectl label configmap ${domainUID}-create-weblogic-sample-domain-job-cm -n $namespace weblogic.resourceVersion=domain-v1 weblogic.domainUID=$domainUID weblogic.domainName=$domainName
```
Sure

doxiao commented on kubernetes/samples/scripts/create-weblogic-domain/domain-home-on-pv/wlst/create-domain.py:
```diff
@@ -0,0 +1,150 @@
+# Copyright 2017, 2018, Oracle Corporation and/or its affiliates.  All rights reserved.
```
Not so sure. If I understand the request correctly, we'll expose about the entire domain topology to the create domain job's Pod env.

doxiao replied:
Okay. I'll change the sample script to use env variables then.

rjeberhard commented:
@doxiao @tbarnes-us @moreaut If comments are addressed, would you please approve the review?  Thanks.

tbarnes-us commented:
@rjeberhard LGTM My comments are about work that should be done, but doesn't need to be in this pull.

moreaut commented:
The image, imagePullSecretName changes look good to me.

I just noticed that this sample doesn't support imagePullPolicy.
I don't think we need to hold up the merge for this, but, should we add it later?

doxiao commented:
> @rjeberhard LGTM My comments are about work that should be done, but doesn't need to be in this pull.

Tom, I believe that I have addressed your comments/suggestions. Can you check?

> The image, imagePullSecretName changes look good to me.
> 
> I just noticed that this sample doesn't support imagePullPolicy.
> I don't think we need to hold up the merge for this, but, should we add it later?

The original create domain job in 1.x does not support imagePullPolicy. It is hardcoded to IfNotPresent. If it is important to support it in the sample, I can add it (either in this PR or the next one).

doxiao commented:
> I just noticed that this sample doesn't support imagePullPolicy.
> I don't think we need to hold up the merge for this, but, should we add it later?

The original create domain job in 1.x does not support imagePullPolicy. It is hardcoded to IfNotPresent. If it is important to support it in the sample, I can add it (either in this PR or the next one).

doxiao commented:
> Hi Dongbo, I think we should add imagePullPolicy eventually - I'm OK if you add it in this PR or in a subsequent one.

Okay. I'll add it in this PR.
